src/SplitTrilinearInterpolation.jl

```diff
@@ -0,0 +1,174 @@
+    vsource::VS
+    "Vertical target grid"
+    vtarget::VT
+    "bilinear object containing the horizontal source and target grids"
+    bilinear::HB
+    "work array, storing fsource on the target horizontal grid"
+    fsourceintermediate::W1
+    "vertical source grid corresponding to target horizontal grid"
+    vsourceintermediate::W2
+end
+
+"""
+    SplitTrilinearHV(
+        (vsource, h1source, h2source),
+        (vtarget, h1target, h2target);
+    )
+
+A constructor for the `SplitTrilinearHV` struct.
+"""
+function SplitTrilinearHV(
+    (vsource, h1source, h2source),
+    (vtarget, h1target, h2target);
+)
+    bilinear = Bilinear(h1source, h2source, h1target, h2target)
+    nvsource = size(vsource, 1)
+    _, _, nh1target, nh2target = get_dims(bilinear)
+    fsourceintermediate = similar(vsource, nvsource, nh1target, nh2target)
+    vsourceintermediate =
+        ndims(vsource) == 1 ? vsource :
+        similar(vsource, nvsource, nh1target, nh2target)
+    ndims(vsource) == 1 ||
+        interpolatebilinear!(vsourceintermediate, bilinear, vsource)
+    return SplitTrilinearHV(
+        vsource,
+        vtarget,
+        bilinear,
+        fsourceintermediate,
+        vsourceintermediate,
+    )
+end
+
+"""
+    split_trilinear_interpolation!(
+        ftarget,
+        split_trilinear::SplitTrilinearHV,
+        fsource;
+        vextrapolate = Flat(),
+        vreverse = false,
+    )
+
+Interpolate the data `fsource` defined on a source grid onto the target grid using split trilinear interpolation.
+This specific function uses the "horizontal first" approach. The following sequence of steps are used for this 
+interpolation:
+
+1). Interpolate `fsource` onto the target horizontal grid, using blinear interpolation, at each of the vertical
+levels. Store the result in `fsourceintermediate` work array.
+
+2). If the column grid is identical for all columns, this step can be skipped and `vsourceintermediate` is identical
+to `vsource`. However, if that is not the case, use bilinear interpolation to compute the `vsourceintermediate` on
+the target horizontal grid, at each level, using bilinear interpolation.
+
+3). Use 1D linear interpolation, for each column, to interpolate `fsourceintermediate` onto the target grid.
+"""
+function split_trilinear_interpolation!(
+    ftarget,
+    split_trilinear::SplitTrilinearHV,
+    fsource;
+    vextrapolate = Flat(),
+    vreverse = false,
+)
+    (; vtarget, bilinear, fsourceintermediate, vsourceintermediate) =
+        split_trilinear
+    # applies bilinear interpolation at each horizontal level
+    interpolatebilinear!(fsourceintermediate, bilinear, fsource)
+    # then apply linear interpolation in the vertical direction
+    interpolate1d!(
+        ftarget,
+        vsourceintermediate,
+        vtarget,
+        fsourceintermediate,
+        Linear(),
+        vextrapolate,
+        reverse = vreverse,
+    )
+    return nothing
+end
+
+struct SplitTrilinearVH{VS, VT, HB, W1}
```

Comment:
docstrings for this structure and its functions would be helpful, as well as noting this is used for vertical-first interpolation (as you have done for `SplitTrilinearHV`)